packages/postgres-database/tests/test_utils_folders_v2.py

```diff
@@ -0,0 +1,1856 @@
+        gid_user_b,
+        source=folder_id_f_user_b,
+        destination=folder_id_f_shared_as_viewer_user_b,
+    )
+
+    # 2. allowed oeprations:
+    # - `USER_A/f_user_a -> f_shared_as_editor_user_a` (& reverse)
+    await _move_back_and_forth(
+        gid_user_a,
+        source=folder_id_f_user_a,
+        destination=folder_id_f_shared_as_editor_user_a,
+        source_parent=folder_id_user_a,
+    )
+    # - `USER_B/f_user_b -> f_shared_as_editor_user_b` (& reverse)
+    await _move_back_and_forth(
+        gid_user_b,
+        source=folder_id_f_user_b,
+        destination=folder_id_f_shared_as_editor_user_b,
+        source_parent=folder_id_user_b,
+    )
+    # - `USER_A/f_user_a -> f_shared_as_owner_user_a` (& reverse)
+    await _move_back_and_forth(
+        gid_user_a,
+        source=folder_id_f_user_a,
+        destination=folder_id_f_shared_as_owner_user_a,
+        source_parent=folder_id_user_a,
+    )
+    # - `USER_B/f_user_b -> f_shared_as_owner_user_b` (& reverse)
+    await _move_back_and_forth(
+        gid_user_b,
+        source=folder_id_f_user_b,
+        destination=folder_id_f_shared_as_owner_user_b,
+        source_parent=folder_id_user_b,
+    )
+
+    # 3. allowed to move in `root` folder
+    for to_move_folder_id, to_move_gid in [
+        (folder_id_f_user_a, gid_user_a),
+        (folder_id_f_user_b, gid_user_b),
+        (folder_id_f_shared_as_owner_user_a, gid_user_a),
+        (folder_id_f_shared_as_owner_user_b, gid_user_b),
+    ]:
+        await folder_move(
+            connection,
+            to_move_folder_id,
+            to_move_gid,
+            destination_folder_id=None,
+        )
+
+    # 4. not allowed to move in `root` folder
+    for to_move_folder_id, to_move_gid in [
+        (folder_id_f_shared_as_editor_user_a, gid_user_a),
+        (folder_id_f_shared_as_editor_user_b, gid_user_b),
+        (folder_id_f_shared_as_viewer_user_a, gid_user_a),
+        (folder_id_f_shared_as_viewer_user_b, gid_user_b),
+        (folder_id_f_shared_as_no_access_user_a, gid_user_a),
+        (folder_id_f_shared_as_no_access_user_b, gid_user_b),
+    ]:
+        with pytest.raises(InsufficientPermissionsError):
+            await folder_move(
+                connection,
+                to_move_folder_id,
+                to_move_gid,
+                destination_folder_id=None,
+            )
+
+    for to_move_gid in [
+        folder_id_f_shared_as_editor_user_a,
+        folder_id_f_shared_as_editor_user_b,
+        folder_id_f_shared_as_viewer_user_a,
+        folder_id_f_shared_as_viewer_user_b,
+        folder_id_f_shared_as_no_access_user_a,
+        folder_id_f_shared_as_no_access_user_b,
+    ]:
+        with pytest.raises(FolderNotSharedWithGidError):
+            await folder_move(
+                connection,
+                folder_id_not_shared,
+                to_move_gid,
+                destination_folder_id=None,
+            )
```

Comment:
this test is 500 lines. is there no way to make more tests with specific purpose?

Reply:
I wonder how as well. I really liked the fact that I have variables that contain exactly what I need and I don't have to mess with keys and dicts.

Reply:
Tried to make ti a bit simpler to read. Let's see if it's better now.